modules/platforms/dotnet/Apache.Ignite.Core/Impl/Binary/BinaryProcessorClient.cs

```diff
@@ -99,11 +105,38 @@ public BinaryType RegisterEnum(string typeName, IEnumerable<KeyValuePair<string,
         }
 
         /** <inheritdoc /> */
-        public string GetTypeName(int id)
+        public string GetTypeName(int id, bool registerSameJavaType)
+        {
+            try
+            {
+                return GetTypeName(id, BinaryProcessor.DotNetPlatformId);
+            }
+            catch (Exception)
```

Comment:
Same as above - please don't use exceptions for control flow. For thin client we can use `errorFunc` parameter of the `DoOutInOp` function to handle errors without throwing exceptions.

Reply:
Fixed.